### PR TITLE
[toplevel] Don't ignore output filename provided by user in -o

### LINKED ISSUE #7447: -o option has stopped working for directories
8.8.0

Linux

Since I updated to coq 8.8.0, the -o option is no longer working like it used to. I am using this option to have a separate build directory, so my Makefile produces a command like coqc -R . Lib -o build/directory/file.vo directory/file.v. This used to work but this is no longer the case. Instead it creates file.vo in the same directory as file.v.

### LINKED ISSUE #7457: #7456 ([toplevel] Don't ignore output filename provided by user in -o) should have a test case in the test-suite
Adding an issue for this as requested by https://github.com/coq/coq/pull/7456#issuecomment-387233510

Probably the thing to do is to find the right category/folder for it, and add a custom `run.sh` script (as in the `coq-makefile` folder and the timing tests therein) to run `coqc` with `-o` on a dummy file and check that it creates a `.vo` file in the right place.

### PR DESCRIPTION
**Pull request by ejgallego**
This was a silly bug introduced in
675a1dc401eb9a5540ba5bc9a522c1f84d4c3d54 that forgot to properly
forward the command line option.

Thanks to @SkySkimmer for finding out the problem.

closes #7447
closes #7457

### CODE REVIEW

**Comment by JasonGross** [comment]
It might be useful to add a test file to the test-suite

**Comment by ejgallego** [comment]
> It might be useful to add a test file to the test-suite

Good idea, I might be short on cycles tho (*) and this fix is important, so please open an issue so we don't forget, or feel free to push to the branch.

(*) I ignore how the test suite works so that'd take me some time.

**Comment by gares** [comment]
I think https://github.com/coq/coq/blob/master/test-suite/misc/exitstatus.sh is a good example to test this functionality.

**Comment by gares** [comment]
I'm preparing the test and pushing it to the branch of @ejgallego

**Comment by gares** [comment]
I should have added this test when `-o` was added. Here it is, not very deep, but have detected the regression.

**Comment by gares** [comment]
I think this fix is to be merged in 8.8, so in any case it has to be checked by @Zimmi48 or @maximedenes ... I guess you guys could also do the review

File: test-suite/misc/coqc_dash_o.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+$coqc misc/coqc_dash_o.v -o /tmp/coqc_dash_o.vo
```
Comment (ejgallego) [inline test-suite/misc/coqc_dash_o.sh]: This is insecure and bad practice, at the very least we should use `mktemp` or a local path. I see we have more stuff like this in the test suite.

**Review by ejgallego — COMMENTED** [review]
Fix insecure file creation.

File: test-suite/misc/coqc_dash_o.sh (hunk 1 above)
Comment (Zimmi48) [reply test-suite/misc/coqc_dash_o.sh]: Interesting! I didn't know about `mktemp`.

Comment (ejgallego) [reply test-suite/misc/coqc_dash_o.sh]: I tried to request changes to my own PR and I discovered I couldn't do it XD

**Comment by gares** [comment]
Hum, while playing with coqc, I tried to pass to it -no-glob, but it still generates a .glob file.

```
gares@gargamel:~/COQ/wt/7456$ touch a.v
gares@gargamel:~/COQ/wt/7456$ bin/coqc -no-glob a
gares@gargamel:~/COQ/wt/7456$ ls
a.glob a.v a.vo
```

@ejgallego does this ring a bell? Otherwise I'll open a separate issue.

**Comment by maximedenes** [comment]
@gares please report the `-no-glob` issue. I did some benchmarking with it not so long ago, I remember being surprised I didn't measure any difference...

**Review by maximedenes — APPROVED** [review]

**Comment by ejgallego** [comment]
From a quick look the logic for `-no-glob` is the same than before, but we could well have a bug as the logic itself is strange.

**Review by Zimmi48 — APPROVED** [review]

**Comment by gares** [comment]
I can't reproduce the no-glob bug anymore, so I guess I was dreaming. Sorry.

**Pull request merged**